enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -252,31 +252,25 @@ impl<'a> Handler<'a> {
     pub fn read(&mut self) -> u64 {
         self.trace("read", 3);
 
-        let fd = self.aex.gpr.rdi.raw();
-        let buf = self.aex.gpr.rsi.raw() as *mut u8;
-        let size = self.aex.gpr.rdx.raw();
-
-        // Allocate some unencrypted memory.
-        let map = match self.ualloc(size) {
-            Err(errno) => return errno as u64,
-            Ok(map) => map,
+        // Do the syscall; replace encrypted memory with unencrypted memory.
+        let res = unsafe {
+            self.proxy(request!(libc::SYS_read => self.aex.gpr.rdi, &self.block.buf, usize::from(self.aex.gpr.rdx)))
```

Comment:
What happens if `self.aex.gpr.rdx` is longer than `self.block.buf.len()`?

Reply:
Nothing good. I've added a check for this. Thanks.